src/H5E.c

```diff
@@ -889,3 +889,114 @@ H5Eappend_stack(hid_t dst_stack_id, hid_t src_stack_id, hbool_t close_source_sta
 done:
     FUNC_LEAVE_API(ret_value)
 } /* end H5Eappend_stack() */
+
+/*-------------------------------------------------------------------------
+ * Function:    H5Eis_paused
+ *
+ * Purpose:     Check if pushing errors on an error stack is paused
+ *
+ * Return:      Non-negative value on success/Negative on failure
+ *
+ *-------------------------------------------------------------------------
+ */
+herr_t
+H5Eis_paused(hid_t stack_id, hbool_t *is_paused)
+{
+    H5E_stack_t *stack;               /* Error stack */
+    herr_t       ret_value = SUCCEED; /* Return value */
+
+    /* Don't clear the error stack! :-) */
+    FUNC_ENTER_API_NOCLEAR(FAIL)
+
+    /* Get the correct error stack */
+    if (stack_id == H5E_DEFAULT) {
+        if (NULL == (stack = H5E__get_my_stack()))
+            HGOTO_ERROR(H5E_ERROR, H5E_CANTGET, FAIL, "can't get current error stack");
+    } /* end if */
+    else {
+        /* Only clear the error stack if it's not the default stack */
+        H5E_clear_stack();
+
+        /* Get the error stack to operate on */
+        if (NULL == (stack = (H5E_stack_t *)H5I_object_verify(stack_id, H5I_ERROR_STACK)))
+            HGOTO_ERROR(H5E_ARGS, H5E_BADTYPE, FAIL, "not an error stack ID");
+    } /* end else */
+
+    /* Check arguments */
+    if (NULL == is_paused)
+        HGOTO_ERROR(H5E_ARGS, H5E_BADVALUE, FAIL, "is_paused parameter is NULL");
+
+    /* Check if the stack is paused */
+    *is_paused = (stack->paused > 0);
+
+done:
+    FUNC_LEAVE_API(ret_value)
+} /* end H5Eis_paused() */
+
+/*-------------------------------------------------------------------------
+ * Function:    H5Epause_stack
+ *
+ * Purpose:     Pause pushing errors on an error stack
+ *
+ * Return:      Non-negative value on success/Negative on failure
+ *
+ *-------------------------------------------------------------------------
+ */
+herr_t
+H5Epause_stack(hid_t stack_id)
+{
+    H5E_stack_t *stack;               /* Error stack */
+    herr_t       ret_value = SUCCEED; /* Return value */
+
+    FUNC_ENTER_API(FAIL)
+
+    /* Applications can't pause the default stack */
+    if (stack_id == H5E_DEFAULT)
```

Comment:
Hmm, I think this sort of comes down to deciding if we want to have multiple methods to essentially disable error reporting, but it seems to me that `H5Epause/resume_stack` on the default error stack is a reasonable use case. Otherwise, it seems like there's a bit of a disconnect where internal code is intended to use the internal `H5E_pause_stack` to temporarily disable error reporting, while "external" code has to use `H5Eget/set_auto`, which developers have to remember. Is there harm in allowing an application to call `H5Epause/resume_stack` on the default error stack and promoting that as a more natural way of temporarily disabling error reporting, as opposed to just temporarily NULLing out the printing callback?

Reply:
I remember thinking "I don't think this will work from user space" when I was writing it, so I disabled it.   But, after your prompt, I went back and think it should be OK.  I think my main concern is that a plugin could manipulate the default error stack's pause status (unpause it, or make it paused when it returns), but that's true for the get_auto/set_auto also.

I'll update the code to allow it, and also switch the H5E_BEGIN_TRY / H5E_END_TRY macros to pause & resume pushing errors on the stack, since that's more logical than just disabling the reporting (as you say).

Reply:
Now that I think about it a little more - I can add some code to save & restore library state around making a call to a plugin's function pointer (starting with the error stack reporting routine and pause status).   We're going to need that concept when we start converting routines that make plugin callbacks to be concurrent, and it's not bad to start now.

Reply:
> I remember thinking "I don't think this will work from user space" when I was writing it, so I disabled it. But, after your prompt, I went back and think it should be OK. I think my main concern is that a plugin could manipulate the default error stack's pause status (unpause it, or make it paused when it returns), but that's true for the get_auto/set_auto also.
> 
> I'll update the code to allow it, and also switch the H5E_BEGIN_TRY / H5E_END_TRY macros to pause & resume pushing errors on the stack, since that's more logical than just disabling the reporting (as you say).

OK, I've pushed an update to allow pausing and resuming the default error stack.   I decided not to change the H5E_BEGIN_TRY / H5E_END_TRY macros, since that would modify application visible behavior.

Reply:
> Now that I think about it a little more - I can add some code to save & restore library state around making a call to a plugin's function pointer (starting with the error stack reporting routine and pause status). We're going to need that concept when we start converting routines that make plugin callbacks to be concurrent, and it's not bad to start now.

I haven't made this change yet, since I want to think about it a little more.  Generally speaking, I'm planning to add the state of the default error stack to the information retrieved and restored with H5CX_retrieve_state / H5CX_restore_state, but I'm not 100% certain yet.  There's no urgency to this defensive state saving, so I've put it on my to-do list to tackle in a little while.